### PR TITLE
Make discovery repository details configurable.

### PR DESCRIPTION
**Pull request by UmanShahzad**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

This just moves variables around a little to make certain things more easily configurable. In particular, this PR is aimed at making the repository configuration for course-discovery configurable.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @UmanShahzad! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?number=4137&repo=edx%2Fconfiguration"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @UmanShahzad! I've created [OSPR-1951](https://openedx.atlassian.net/browse/OSPR-1951) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: playbooks/roles/discovery/defaults/main.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@
 # Defaults for role discovery
 #
 
+DISCOVERY_GIT_IDENTITY: !!null
```
Comment (haikuginger) [inline playbooks/roles/discovery/defaults/main.yml:14]: Why is this `!!null` here, while it's just `null` in the edx_django_service` defaults?

**Review by haikuginger — APPROVED** [review]
:+1:, with one question, and conditional on successful use of the code to deploy PearsonX.

- [x] I tested this: Validated that the changes made sense in the context of rendering the code for course discovery more configurable, and that minimal behavioral changes should occur as a result.
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~
- [x] ~~Includes documentation~~

File: playbooks/roles/discovery/defaults/main.yml (hunk 1 above)
Comment (UmanShahzad) [reply playbooks/roles/discovery/defaults/main.yml:14]: I'm not fully sure myself: I copied this from ECOMMERCE_GIT_IDENTITY. Ecommerce is using `edx_service`, and `edx_django_service` itself uses `edx_service`. Reading the difference between `null` and `!!null` in docs isn't really clarifying anything for me.

Comment (haikuginger) [reply playbooks/roles/discovery/defaults/main.yml:14]: Okay sounds good.

**Comment by UmanShahzad** [comment]
@haikuginger This is now working properly on PearsonX's deploy. I was able to capture the repository `https://github.com/open-craft/course-discovery.git` by modifying the new `DISCOVERY_REPOS` setting in this PR, and the branch `opencraft-release/ginkgo.1-pearson`, which contains some cherry-picked commits on top of `open-release/ginkgo.1`.

CC @feanil for review

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,10 @@
 - Role: discovery
+  - Added `DISCOVERY_REPOS` to allow configuring discovery repository details.
+
+- Role: edx_django_service
+  - Made `edx_django_service_repos` fully configurable.
```
Comment (feanil) [inline CHANGELOG.md]: This is a very vague statement, can you be more specific about what you changed?

Comment (UmanShahzad) [reply CHANGELOG.md]: "Made keys of `edx_django_service_repos` all individually configurable"? Does that sound good, @feanil ?

Comment (feanil) [reply CHANGELOG.md]: That sounds good.  I think you can also say which new keys are now configurable.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,10 @@
 - Role: discovery
+  - Added `DISCOVERY_REPOS` to allow configuring discovery repository details.
+
+- Role: edx_django_service
+  - Made the keys `edx_django_service_git_protocol`, `edx_django_service_git_domain`, and `edx_django_service_git_path` of `edx_django_service_repos` all individually configurable.
```
Comment (UmanShahzad) [inline CHANGELOG.md:5]: @feanil Looks good?

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@UmanShahzad 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/configuration&entry.867055334=https://github.com/edx/configuration/pull/4137&entry.1484655318=https://github.com/UmanShahzad&entry.752974735=2017-10-07+01:06&entry.1917517419=2017-10-11+15:51&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**